src/main/kotlin/org/neo4j/graphql/handler/filter/OptimizedFilterHandler.kt

```diff
@@ -260,15 +262,32 @@ class OptimizedFilterHandler(val type: GraphQLFieldsContainer) {
                 variablePrefix: String,
                 conditions: List<Predicate<FieldOperator>>,
                 filterParams: MutableMap<String, Any?>,
+                appendObjectFieldName: Boolean,
                 conditionAdder: (where: WithWhere?, condition: Condition) -> WithWhere
         ): WithWhere? {
             var where: WithWhere? = null
             for (conditionField in conditions) {
-                val prop = propertyContainer.property(conditionField.fieldDefinition.name)
-                val parameter = Cypher.parameter(variablePrefix + "_" + conditionField.queryField.name)
-                val condition = conditionField.op.conditionCreator(prop, parameter)
-                filterParams[parameter.name] = conditionField.queryField.value.toJavaValue()
-                where = conditionAdder(where, condition)
+                if (conditionField.fieldDefinition.type.isNeo4jType()
```

Comment:
we have a lot of places where larger chunks of code related to spatial / distance are now spread through the code-base like here, would be there any way of extracting them/putting them together so that the implementation doesn't leak into all these places?